var/spack/repos/builtin/packages/py-hail/package.py

```diff
@@ -0,0 +1,234 @@
+    # url = "https://github.com/hail-is/hail/archive/refs/tags/0.2.130.tar.gz"
+
+    maintainers("teaguesterling")
+    license("MIT", checked_by="teaguesterling")
+
+    version("0.2.132", commit="678e1f52b9999cb05ebf03fd360e5c4506bd6dad")
+    version("0.2.131", commit="11d9b2ff89da9ef6a4f576be89f1f06959580ea4")
+    version("0.2.130", commit="bea04d9c79b5ca739364e8c121132845475f617a")
+    version("0.2.129", commit="41126be2df04e4ef823cefea40fba4cadbe5db8a")
+
+    resource(
+        name="catch",
+        url="https://github.com/catchorg/Catch2/releases/download/v2.6.0/catch.hpp",
+        sha256="a86133b34d4721b6e1cf7171981ea469789f83f2475907b4033012577e4975fe",
+        destination="hail/src/main/resources/include/catch.hpp",
+        expand=False,
+    )
+
+    resource(
+        name="libsimdpp-2.1",
+        extension="tar.gz",
+        url="https://storage.googleapis.com/hail-common/libsimdpp-2.1.tar.gz",
+        sha256="b0e986b20bef77cd17004dd02db0c1ad9fab9c70d4e99594a9db1ee6a345be93",
+        destination="hail/src/main/c",
+    )
+
+    resource(
+        name="mill-0.11.7",
+        url="https://repo1.maven.org/maven2/com/lihaoyi/mill-dist/0.11.7/mill-dist-0.11.7.jar",
+        sha256="278b430150af899495d360d1f886e223e78bb4a20e67144a240bfb7e2d4f6085",
+        destination="hail/mill",
+        expand=False,
+    )
+
+    variant("native", default=True, description="Compile native HAIL backend")
+    variant(
+        "query_backend",
+        values=["undefined", "spark", "batch"],
+        default="spark",
+        description="Configure HAIL query backend at build",
+    )
+
+    depends_on("python@3.9:", type=("build", "run"))
+    depends_on("py-pip", type="build")
+    depends_on("py-wheel", type="build")
+    depends_on("py-build@1.1+virtualenv", type="build", when="@0.2.131:")
+    depends_on("c", type="build")
+    depends_on("cxx", type="build")
+
+    # HAIL bundle is tied to specific runtime versions
+    # HAIL spec, Java sec, Spark spec, Scala spec
+    # We're not accurately capturing previous versions
+    for hail, java, spark, scala in [
+        # 0.2.130 and before (to somwhere around 0.2.64) used Spark 3.3
+        # And either Java 8 or Java 11
+        (":0.2.130", "8,11", "3.3", "2.12"),
+        # 0.2.131 updated to Java 11 and Spark 3.5
+        # Undocumented bump was to scala 2.12.13 for scala.annotation.noerror
+        ("0.2.131:", "11", "3.5", "2.12.18:2.12"),
+    ]:
+        with default_args(type=("build", "run"), when=f"@{hail}"):
+            depends_on(f"java@{java}")
+            depends_on(f"scala@{scala}")
+            depends_on(f"spark@{spark}")
+            # This should match spark but isn't actually enforced
+            # by the PySpark package and they can conflit.
+            depends_on(f"py-pyspark@{spark}")
+
+    with default_args(type=("build", "run")):
+        # Hail build requirements
+        depends_on("blas")
+        depends_on("lapack")
+        depends_on("lz4")
```

Comment:
How comes these three deps are `build, run` instead of `build, link` ?

Reply:
Whoops!